internal/edgedbtypes/datetime.go

```diff
@@ -660,6 +660,26 @@ func (d Duration) String() string {
 	return strings.Join(buf, "")
 }
 
+// AsNanoseconds returns [time.Duration] represented as nanoseconds,
+// after transforming from Duration microsecond representation.
+// Returns an error if the Duration is too long and would cause an overflow of
+// the internal int64 representation.
+func (d Duration) AsNanoseconds() (time.Duration, error) {
+	if int64(d) > math.MaxInt64/int64(time.Microsecond) ||
+		int64(d) < math.MinInt64/int64(time.Microsecond) {
+		return time.Duration(0), fmt.Errorf(
+			"Duration is too large to be represented as nanoseconds",
+		)
+	}
+	return time.Duration(d) * time.Microsecond, nil
+}
+
+// FromNanoseconds creates a Duration represented as microseconds
+// from a [time.Duration] represented as nanoseconds.
+func FromNanoseconds(d time.Duration) Duration {
```

Comment:
I think this needs to be added to [export.go](https://github.com/edgedb/edgedb-go/blob/master/export.go) to be publicly accessible.

Reply:
Thanks! I also changed the name for clarity